server/index.spec.js

```diff
@@ -1,640 +1,690 @@
+        .set('cookie', cookie)
+        .expect(function(res) {
+          expect(500).to.be.eql(res.statusCode);
+        });
+    });
+  });
+
+  describe('GET /doppler-lists', function() {
+    it('Should return the Doppler lists correctly', async function() {
+      this.sandbox
+        .stub(mocks.wrappedRedisClient, 'hgetall')
+        .callsFake((key, cb) => {
+          cb(null, { accessToken, dopplerAccountName, dopplerApiKey });
+        });
+
+      fetchStub
+        .withArgs(
+          `https://restapi.fromdoppler.com/accounts/${querystring.escape(
+            dopplerAccountName
+          )}/lists?page=1&per_page=200&state=active`,
+          {
+            headers: { Authorization: `token ${dopplerApiKey}` },
+          }
+        )
+        .returns(
+          Promise.resolve({
+            status: 200,
+            json: async function() {
+              return dopplerApiResponses.LISTS_PAGE_RESULT_200;
+            },
+          })
+        );
+
+      await request(app)
+        .get('/doppler-lists')
+        .set('cookie', cookie)
+        .expect(function(res) {
+          expect(
+            '{"items":[{"listId":1459381,"name":"shopify"},{"listId":1222381,"name":"marketing"},{"listId":1170501,"name":"development"}],"itemsCount":3}'
+          ).to.be.eql(res.text);
+          expect(200).to.be.eql(res.statusCode);
+        });
+    });
+
+    it('Should return 500 status code when there is not a Doppler API key', async function() {
+      this.sandbox
+        .stub(mocks.wrappedRedisClient, 'hgetall')
+        .callsFake((key, cb) => {
+          cb(null, { accessToken });
+        });
+
+      await request(app)
+        .get('/doppler-lists')
+        .set('cookie', cookie)
+        .expect(function(res) {
+          expect(500).to.be.eql(res.statusCode);
+        });
+    });
+  });
+
+  describe('POST /create-doppler-list', function() {
+    it('Should create the Doppler list correctly', async function() {
+      fetchStub
+        .withArgs(
+          `https://restapi.fromdoppler.com/accounts/${querystring.escape(
+            dopplerAccountName
+          )}/lists`,
+          {
+            body: '{"name":"Fresh List"}',
+            method: 'POST',
+            headers: { Authorization: `token ${dopplerApiKey}` },
+          }
+        )
+        .returns(
+          Promise.resolve({
+            status: 201,
+            json: async function() {
+              return dopplerApiResponses.LIST_CREATED_201;
+            },
+          })
+        );
+
+      this.sandbox
+        .stub(mocks.wrappedRedisClient, 'hgetall')
+        .callsFake((key, cb) => {
+          cb(null, { accessToken, dopplerAccountName, dopplerApiKey });
+        });
+
+      await request(app)
+        .post('/create-doppler-list')
+        .send({ name: 'Fresh List' })
+        .set('cookie', cookie)
+        .expect(function(res) {
+          expect(res.body).to.be.eql({ listId: '1462409' });

```

Comment:
I think that this listId should be requested to API Doppler with the request GET /accounts/{accountName}/lists/{listId}

Reply:
Why should we request the just created resource? Sorry I don't see your point.

Reply:
I don't remember if the response of the POST request return the listId value if not is necessary send a GET request to API Doppler to return listId 
![image](https://user-images.githubusercontent.com/6796523/47654831-702fd780-db6a-11e8-97cb-932e304970f7.png)

and avoid the hardcode value

Reply:
Oohh,  I see. Yes the API is returning the Id of the just created resource. In this test we're mocking the server response so there is not need to consume any service.